### DISCUSSION TITLE
How to live render only a section of the console with Python Rich package

### DISCUSSION THREAD

**Discussion by Yogesh7920** [Textualize/rich#3137] [Q&A]
In the official example provided in [layout.py](https://github.com/Textualize/rich/blob/master/examples/layout.py), the entire layout is being rendered four times every second, but it's only the header that is changing, thus I want to have Live only for the header section, how can I do that.

I tried the following

```
layout["header"].update(Clock())

with Live(layout["header"], redirect_stderr=False, auto_refresh=False, console=console) as live:
    console.print(layout["main"])
    try:
        while True:
            live.refresh()
            sleep(1)
    except KeyboardInterrupt:
        pass
```

But even in this case, the main is getting printed before the header itself.

**Comment by willmcgugan** [comment]
That is correct. That is how Live works. If you want something more dynamic, consider [Textual](https://textual.textualize.io/).
